tools/check_format.py

```diff
@@ -164,13 +164,24 @@
 # yapf: enable
 
 
-# Map a line transformation function across each line of a file.
-# .bak temporaries.
-def replaceLines(path, line_xform):
+# Map a line transformation function across each line of a file,
+# writing the result lines as requested.
+def evaluateLines(path, line_xform, write=True):
+  format_flag = True
+  output_lines = []
+  for line_number, line in enumerate(readLines(path)):
+    if line.find("// clang-format on") != -1:
```

Comment:
This is a really nice touch. But why `// clang-format`? IIUC, this applies not to the clang formatter, but to the custom 1-line-at-a-time source hacking done in Python.

I guess the point of this is that we'll just have a single control to enable/disable some of the format checks; is that right? Maybe just add a comment?

Also WDYT of printing a warning if the clang-format directives are unbalanced or nested?

Reply:
See short discussion of https://envoyproxy.slack.com/archives/C78HA81DH/p1579280748007800

Some // envoy-lint off custom flag?

Off-balance is fine (// flag off .... remainder of file), and there is no harm in the converse (// flag on ... [the default]).  Should we add a nesting counter? That's doable.

The issue is that these custom 1-liners were all well and good 99% of the time, but simple tables formatted for legibility, and other exceptional comments become impossible.

Reply:
I'm happy leaving this as clang-format even for lint checks that are not related to clang. That way we can have a single keep-out directive. Please doc that both a python code comment and also in STYLE.md

I'm not sure about the nesting semantics, so I'd have check_format report an error if there is:
 * on while in 'on' state
 * off while in 'off' state
 * EOF while in 'on' state

Reply:
Confirmed, all off's are matched by on's in source and test already, so implementing as 
suggested.